src/main/java/com/qudini/api/requests/composition/Merchants.java

```diff
@@ -0,0 +1,174 @@
+                        envBaseUri,
+                        qudiniAppUsername,
+                        qudiniAppPassword);
+
+            }
+        } catch (IOException e) {
+            log.error(String.format("There was a problem with the stated csv file or filepath: %s", merchantsFilePath));
+            throw e;
+        }
+
+
+    }
+
+    public void createMerchant(
+            String name,
+            String maxVenues,
+            String templateKey,
+            String contractStatusKey,
+            String industrySectorKey,
+            String sizeCategoryKey,
+            String salesRegionKey,
+            String countryKey,
+            String timezoneKey,
+            String languageKey,
+            String billingTypeKey,
+            String salesAssigneeKey,
+            String reportWalkoutThreshold,
+            String envBaseUri,
+            String qudiniAppUsername,
+            String qudiniAppPassword)
+            throws UnsupportedEncodingException {
+
+        List<NameValuePair> paramsAsNameValuePairList = new ArrayList<>();
+
+        paramsAsNameValuePairList.add(new BasicNameValuePair(MERCHANT_CSV_HEADER_NAME, name));
+        paramsAsNameValuePairList.add(new BasicNameValuePair(MERCHANT_CSV_HEADER_MAX_VENUES, maxVenues));
+        paramsAsNameValuePairList.add(new BasicNameValuePair(MERCHANT_CSV_HEADER_TEMPLATE_KEY, templateKey));
+        paramsAsNameValuePairList.add(new BasicNameValuePair(MERCHANT_CSV_HEADER_CONTRACT_STATUS_KEY, contractStatusKey));
+        paramsAsNameValuePairList.add(new BasicNameValuePair(MERCHANT_CSV_HEADER_INDUSTRY_SECTOR_KEY, industrySectorKey));
+        paramsAsNameValuePairList.add(new BasicNameValuePair(MERCHANT_CSV_HEADER_SIZE_CATEGORY_KEY, sizeCategoryKey));
+        paramsAsNameValuePairList.add(new BasicNameValuePair(MERCHANT_CSV_HEADER_SALES_REGION_KEY, salesRegionKey));
+        paramsAsNameValuePairList.add(new BasicNameValuePair(MERCHANT_CSV_HEADER_COUNTRY_KEY, countryKey));
+        paramsAsNameValuePairList.add(new BasicNameValuePair(MERCHANT_CSV_HEADER_TIMEZONE_KEY, timezoneKey));
+        paramsAsNameValuePairList.add(new BasicNameValuePair(MERCHANT_CSV_HEADER_LANGUAGE_KEY, languageKey));
+        paramsAsNameValuePairList.add(new BasicNameValuePair(MERCHANT_CSV_HEADER_BILLING_TYPE_KEY, billingTypeKey));
+        paramsAsNameValuePairList.add(new BasicNameValuePair(MERCHANT_CSV_HEADER_SALES_ASSIGNEE_KEY, salesAssigneeKey));
+        paramsAsNameValuePairList.add(new BasicNameValuePair(MERCHANT_CSV_HEADER_REPORT_WALKOUT_THRESHOLD, reportWalkoutThreshold));
+
+
+        createMerchant(paramsAsNameValuePairList, envBaseUri, qudiniAppUsername, qudiniAppPassword);
+
+    }
+
+    public void createMerchant(List<NameValuePair> merchantNameValuePairs,
+                               String envBaseUri,
+                               String qudiniAppUsername,
+                               String qudiniAppPassword) throws UnsupportedEncodingException {
```

Comment:
What do you think of using the token instead of (user, pass) and include it only in the request sender where you are adding the headers? You could also include the token in the config instead of (user, pass).

Reply:
> You could also include the token in the config instead of (user, pass).

It may be done, but the reasons behind I not doing that was:
- although token is static generated, it still needs to be generated (even if it is only one time for each user/pass), if I added it to the configuration we probably should have a "service" user to use is token instead of this admin user
- user/pass allow anyone to user his user to perform any experience and not being "locked" to some user/pass or token. Also it may be extended in the future to be able to get those from cmd line

> What do you think of using the token instead of (user, pass) and include it only in the request sender where you are adding the headers? 

I may not fully understand what you mean here, but if I understood correctly my intent was to not "access" the generate token directly from the "runner" (although not it is not done) so only the classes that would extend RequestSender would be able to generate the token. That is why all of them get the user/pass instead of the token.
Again if you see that may be "too much complication" for this type of data-generation I completely trust your judgement and will change the classes so the token is either generated in the runner and passed as parameter or already be placed on the config file.

Reply:
If the pair is always read from the config file I would read it directly in the RequestSender, generate the token there and pass it in every request. Consumers of RequestSender would not need to know about user/pass or token and no need to pass them in every method. Or have a RequestSender init method to init user/pass and baseuri as they don't change after being set.

Reply:
Yes seems a good idea, I will do that refactor. 
Thank you for pointing it out.